src/silo/query_engine/filter_expressions/string_search.cpp

```diff
@@ -0,0 +1,113 @@
+#include "silo/query_engine/filter_expressions/string_search.h"
+
+#include <optional>
+#include <utility>
+
+#include <fmt/format.h>
+#include <re2/re2.h>
+#include <nlohmann/json.hpp>
+
+#include "silo/common/string.h"
+#include "silo/query_engine/filter_expressions/expression.h"
+#include "silo/query_engine/operators/bitmap_producer.h"
+#include "silo/query_engine/query_parse_exception.h"
+#include "silo/storage/database_partition.h"
+
+namespace silo {
+class Database;
+namespace query_engine::operators {
+class Operator;
+}  // namespace query_engine::operators
+}  // namespace silo
+
+namespace silo::query_engine::filter_expressions {
+
+StringSearch::StringSearch(std::string column, std::unique_ptr<re2::RE2> search_expression)
+    : column(std::move(column)),
+      search_expression(std::move(search_expression)) {}
+
+std::string StringSearch::toString() const {
+   return fmt::format("column {} regex_matches \"{}\"", column, search_expression->pattern());
+}
+
+namespace {
+template <typename GenericStringColumn>
+std::unique_ptr<silo::query_engine::operators::Operator> createMatchingBitmap(
+   const GenericStringColumn& string_column,
+   const RE2& search_expression,
+   size_t row_count
+) {
+   return std::make_unique<operators::BitmapProducer>(
+      [&, row_count]() {
+         roaring::Roaring result_bitmap;
+         for (size_t row_idx = 0; row_idx < row_count; ++row_idx) {
+            const auto& embedded_value = string_column.getValues().at(row_idx);
+            const auto& string_value = string_column.lookupValue(embedded_value);
+            if (re2::RE2::FullMatch(string_value, search_expression)) {
+               result_bitmap.add(row_idx);
+            }
+         }
+         return OperatorResult(std::move(result_bitmap));
+      },
+      row_count
+   );
+}
+
+}  // namespace
+
+std::unique_ptr<silo::query_engine::operators::Operator> StringSearch::compile(
+   const silo::Database& /*database*/,
+   const silo::DatabasePartition& database_partition,
+   Expression::AmbiguityMode /*mode*/
```

Comment:
Is this an unused parameter of a shared interface, or forgotten, or for the future? (Add comment?)